nifi-extension-bundles/nifi-standard-bundle/nifi-standard-rules/src/main/java/org/apache/nifi/flowanalysis/rules/RestrictBackpressureSettings.java

```diff
@@ -0,0 +1,242 @@
+            .description("This is the maximum value that should be set for the Data Size backpressure setting on connections. "
+                    + "This can be used to prevent a user from setting a very high value that may be filling up the content repo.")
+            .required(true)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .defaultValue("1 GB")
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            COUNT_MIN,
+            COUNT_MAX,
+            SIZE_MIN,
+            SIZE_MAX
+    );
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> results = new ArrayList<ValidationResult>();
+
+        final long minCount = validationContext.getProperty(COUNT_MIN).asLong();
+        final long maxCount = validationContext.getProperty(COUNT_MAX).asLong();
+        final double minSize = validationContext.getProperty(SIZE_MIN).asDataSize(DataUnit.B);
+        final double maxSize = validationContext.getProperty(SIZE_MAX).asDataSize(DataUnit.B);
+
+        if (minCount > maxCount) {
+            results.add(
+                    new ValidationResult.Builder()
+                    .subject(COUNT_MIN.getName())
+                    .valid(false)
+                    .explanation("Value of '" + COUNT_MIN.getName() + "' cannot be strictly greater than '" + COUNT_MAX.getName() + "'")
+                    .build());
+        }
+        if (Double.compare(minSize, maxSize) > 0) {
+            results.add(
+                    new ValidationResult.Builder()
+                    .subject(SIZE_MIN.getName())
+                    .valid(false)
+                    .explanation("Value of '" + SIZE_MIN.getName() + "' cannot be strictly greater than '" + SIZE_MAX.getName() + "'")
+                    .build());
+        }
+
+        return results;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public Collection<GroupAnalysisResult> analyzeProcessGroup(VersionedProcessGroup pg, FlowAnalysisRuleContext context) {
+        final Collection<GroupAnalysisResult> results = new HashSet<GroupAnalysisResult>();
+
+        final long minCount = context.getProperty(COUNT_MIN).asLong();
+        final long maxCount = context.getProperty(COUNT_MAX).asLong();
+        final double minSize = context.getProperty(SIZE_MIN).asDataSize(DataUnit.B);
+        final double maxSize = context.getProperty(SIZE_MAX).asDataSize(DataUnit.B);
+
+        // Map of all id/components to generate more human readable violations
+        final Map<String, VersionedComponent> idComponent = Stream.of(
+                    pg.getFunnels().stream(),
+                    pg.getProcessors().stream(),
+                    pg.getInputPorts().stream(),
+                    pg.getOutputPorts().stream()
```

Comment:
I haven't thought it through fully, but is there a reason pg.getProcessGroups() and pg.getRemoteProcessGroups() are not included here?

Reply:
After looking at this more (while building a new rule myself), do you need this map?  It seems that a source and destination ConnectableComponent, available from the VersionedConnection itself, has the component's id, name and type available.

Reply:
I don't have PGs and RPGs because connections are not connecting to PGs/RPGs but to the ports.

I started making the changes as you suggested (just use the ConnectableComponent objects) but the problem is with `GroupAnalysisResult.forComponent` that expects a VersionedComponent. After a quick look, I don't think we would want to change this logic. If we were only building the violations on the connection itself, what you suggest would work but if we can make a component invalid when the rule has an enforced policy, we should do it.

Reply:
Thanks for the explanation @pvillard31, this makes sense.

While working on another FlowAnalysisRule, I may spend some time trying to optimize this to avoid creating this map until a violation is detected.  With many ProcessGroups to analyze, I would expect the vast majority to not have any violations.